src/rag_processor/websocket/bridge.py

```
@@ -0,0 +1,199 @@
+                    break
+                logger.warning(
+                    "Event bridge listener error; reconnecting",
+                    error=str(e),
+                    retry_in_seconds=backoff,
+                )
+                # Drop the broken connection so the next iteration re-subscribes.
+                await self._cleanup()
+                await asyncio.sleep(backoff)
+                backoff = min(backoff * 2, self._max_backoff)
+
+    async def _relay(self, raw: Any) -> None:
```

Ruff (0.15.14)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `raw`

(ANN401)